www/docs/customization/dockers_v2.md

```diff
@@ -0,0 +1,199 @@
+    # directory, so if you add `foo/bar.json` here, on your Dockerfile you can
+    # `COPY foo/bar.json /whatever.json`.
+    # Also note that the paths here are relative to the directory in which
+    # GoReleaser is being run (usually the repository root directory).
+    # This field does not support wildcards, you can add an entire directory here
+    # and use wildcards when you `COPY`/`ADD` in your Dockerfile.
+    extra_files:
+      - config.yml
+
+    # Labels to be added to the image.
+    labels:
+      "org.opencontainers.image.description": "My software"
+      "org.opencontainers.image.created": "{{.Date}}"
+      "org.opencontainers.image.name": "{{.ProjectName}}"
+      "org.opencontainers.image.revision": "{{.FullCommit}}"
+      "org.opencontainers.image.version": "{{.Version}}"
+      "org.opencontainers.image.source": "{{.GitURL}}"
+
+    # Platforms to build.
+    #
+    # Default: [ linux/amd64 linux/arm64 ]
+    platforms:
+      - linux/amd64
+      - linux/arm64
+
+    # Additional `--build-arg`s to be passed.
+    #
+    # Templates: allowed.
+    build_args:
+      FOO: bar
+
+    # Retry configuration.
+    retry:
+      # Attempts of retry.
+      #
+      # Default: 10.
+      attempts: 5
+
+      # Delay between retry attempts.
+      #
+      # Default: 10s.
+      delay: 5s
+
+      # Maximum delay between retry attempts.
+      #
+      # Default: 5m.
+      max_delay: 2m
+```
+
+!!! important "dockers_v2"
+
+    The `dockers_v2` name is provisional.
+
+    It will replace `dockers` and `docker_manifests` in GoReleaser v3 (no ETA),
+    and will be called only `dockers`.
+
+    It is been done this way to prevent breaking changes in feature releases.
+
+<!-- md:templates -->
+
+## How it works
+
+You can declare multiple Docker images.
+They will be matched against the binaries generated by your `builds` section and
+packages generated by your `nfpms` section.
+
+If you have only one item in the `builds` list,
+the configuration can be as easy as adding the
+name and tags of your images to your `.goreleaser.yaml` file:
+
+```yaml
+dockers_v2:
+  - images:
+      - user/repo
+```
+
+You also need to create a `Dockerfile` in your project's root directory:
+
+```dockerfile
+FROM scratch
+ARG TARGETPLATFORM
+ENTRYPOINT ["/usr/bin/myprogram"]
+COPY $TARGETPLATFORM/myprogram /usr/bin/
+```
+
+This configuration will build and push a Docker image named `user/repo:tagname`.
+
+### The Docker build context
+
+Note that we are not building any go files in the Docker build phase, we are
+merely copying the binary to a `scratch` image and setting up the `entrypoint`.
+
+The idea is that you reuse the previously built binaries instead of building
+them again when creating the Docker image.
+
+The build context itself is a temporary directory which contains the binaries
+and packages for the target platforms, which you can `COPY` into your image
+(mind the use of `$TARGETPLATFORM` above).
```

Comment:
Edge Case scenario:

ARM32 builds (for older RPIs) `GOARCH=arm GOARM=7` have `TARGETPLATFORM=linux/arm/v7`. I'm not checked the code here, but I would like to keep this in your mind.